homeassistant/components/google_assistant/trait.py

```diff
@@ -1316,18 +1316,33 @@ def query_attributes(self):
     async def _execute_volume_relative(self, data, params):
         # This could also support up/down commands using relativeSteps
         relative = params["volumeRelativeLevel"]
-        current = self.state.attributes.get(media_player.ATTR_MEDIA_VOLUME_LEVEL)
 
-        await self.hass.services.async_call(
-            media_player.DOMAIN,
-            media_player.SERVICE_VOLUME_SET,
-            {
-                ATTR_ENTITY_ID: self.state.entity_id,
-                media_player.ATTR_MEDIA_VOLUME_LEVEL: current + relative / 100,
-            },
-            blocking=True,
-            context=data.context,
-        )
+        # if we have access to current volume level, do a single 'set' call
+        if media_player.ATTR_MEDIA_VOLUME_LEVEL in self.state.attributes:
+            current = self.state.attributes.get(media_player.ATTR_MEDIA_VOLUME_LEVEL)
+
+            await self.hass.services.async_call(
+                media_player.DOMAIN,
+                media_player.SERVICE_VOLUME_SET,
+                {
+                    ATTR_ENTITY_ID: self.state.entity_id,
+                    media_player.ATTR_MEDIA_VOLUME_LEVEL: current + relative / 100,
+                },
+                blocking=True,
+                context=data.context,
+            )
+        # otherwise do multiple 'up' or 'down' calls
+        else:
+            for x in range(abs(relative)):
+                await self.hass.services.async_call(
```

Comment:
I'm not a python coder so I don't know what `await` is doing here or if it's bad that it's being used in a loop